docstore/gcpfirestore/fs.go

```diff
@@ -554,31 +554,44 @@ func (c *collection) doCommitCall(ctx context.Context, call *commitCall, errs []
 		}
 		return
 	}
-	_ = wrs
 	// Set the revision fields of the documents.
 	// The actions and writes may not correspond, because Update actions may require
 	// two writes. We can tell which writes correspond to actions by the type of write.
 	j := 0
-	for i, wr := range wrs {
-		if _, ok := call.writes[i].Operation.(*pb.Write_Transform); !ok {
-			a := call.actions[j]
-			if a.Doc.HasField(c.opts.RevisionField) {
-				if err := a.Doc.SetField(c.opts.RevisionField, wr.UpdateTime); err != nil {
-					errs[a.Index] = err
-				}
+	for i, a := range call.actions {
+		wr := wrs[j]
+		if a.Doc.HasField(c.opts.RevisionField) {
+			if err := a.Doc.SetField(c.opts.RevisionField, wr.UpdateTime); err != nil {
+				errs[a.Index] = err
 			}
-			j++
 		}
-	}
-	// Set new names for Create actions.
-	for i, a := range call.actions {
-		if call.newNames[i] != "" {
+		if call.newNames[i] != "" && c.nameField != "" {
```

Comment:
It could be empty when the user uses `nameFunc`, there is an issue #2593 to add test for that.

Reply:
That should be checked [here](https://github.com/google/go-cloud/blob/master/docstore/gcpfirestore/fs.go#L227)
which is called from [here](https://github.com/google/go-cloud/blob/master/docstore/docstore.go#L370).

Do you have a case that fails? If so, I'd like to understand how. If not, then just leave a comment here that nameFIeld can't be empty because it's checked in the Key method.

Reply:
That's right, it should not be empty since we only generate new names when there is a nameField.